### PR TITLE
T376320: Part 1

### PR DESCRIPTION
**Pull request by l-olson1214**
**Phabricator:** 
https://phabricator.wikimedia.org/T376320

### Notes
* Grabs data, will send updated slides if available

### Test Steps
1. Using account that has edits on Italian/French, set yourself up to access the YIR entry point
2. Go through slides
3. You should see your amount of edits on slides (if you have any)

### CODE REVIEW

File: WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift [hunk 1]
```diff
@@ -292,4 +293,84 @@ public class WMFYearInReviewDataController {
         
         return true
     }
+    
+    public func fetchUserContributionsCount(username: String, languageCode: String, completion: @escaping (Result<(Int, Bool), Error>) -> Void) {
```
Comment (tonisevener) [inline WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift:297]: @l-olson1214 For this Part 1 PR, can you only include this method and remove the other changes? I'm going for a different approach where the actual data will be fetched upon app resume (see [here](https://github.com/wikimedia/wikipedia-ios/pull/5024/files#diff-6eda737a0f8aa712d064e24a5159ac83bd352538048c5fbe777ded61e08d082dR617) in my saved article personalized slide [PR](https://github.com/wikimedia/wikipedia-ios/pull/5024)) and saved to a persisted report object. We will then reference that object when we create the `YearInReviewSlideContent`, so there won't be any need for marking the slide view models as `@Published` and updating them in the coordinator.

Once my PR is merged, you'll be unblocked to do parts 2 and 3 of this task.
You will be able to call this method from [here](https://github.com/wikimedia/wikipedia-ios/pull/5024/files#diff-1c01e90351b11ab077c5dff447720470dad032889b5206d5e9c5e4a69065649cR186), update and save the report object, then reference and create the slide [here](https://github.com/wikimedia/wikipedia-ios/pull/5024/files#diff-72caf34090dc13194bbfc66eeb449b72948cef8bd240c2a17c71437894632f90R109).

**Review by tonisevener — CHANGES_REQUESTED** [review]

File: WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift (hunk 1 above)
Comment (l-olson1214) [reply WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift:297]: 🫡 Done and done :)

**Review by tonisevener — CHANGES_REQUESTED** [review]
Looking good, just a few more things!

File: WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift [hunk 2]
```diff
@@ -292,4 +293,84 @@ public class WMFYearInReviewDataController {
...
+            completion(.failure(WMFDataControllerError.failureCreatingRequestURL))
+            return
+        }
+        
+        let request = WMFMediaWikiServiceRequest(url: url, method: .GET, backend: .mediaWiki, parameters: parameters)
+        
+        service.performDecodableGET(request: request) { (result: Result<UserContributionsAPIResponse, Error>) in
+            switch result {
+            case .success(let response):
+                let editCount = response.query?.usercontribs.count
```
[the remark this answers is not in the document]
Comment (l-olson1214) [reply WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift]: Great catch! Thanks :)

File: WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift [hunk 3]
```diff
@@ -292,4 +293,84 @@ public class WMFYearInReviewDataController {
...
+            "ucend": ucEndDate,
+            "ucuser": username,
+            "ucnamespace": "0",
+            "ucprop": "ids|title|timestamp|tags|flags"
+        ]
+
+        guard let url = URL(string: "https://\(languageCode).wikipedia.org/w/api.php") else {
+            completion(.failure(WMFDataControllerError.failureCreatingRequestURL))
+            return
+        }
```
[the remark this answers is not in the document]
Comment (l-olson1214) [reply WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift]: I originally had that but couldn't figure out how to actually GET the project where I was calling it so I did this - this is much nicer!

**Review by tonisevener — APPROVED** [review]

**Pull request merged**